### DISCUSSION TITLE
Bootstrapping API not implemented by Resteasy Reactive

### DISCUSSION THREAD

**Discussion by sverkera** [quarkusio/quarkus#34780] [Q&A]
I am using NettyJaxrsServer in my testcases to instanciate mock servers for my jaxrs api's. This works great in legacy resteasy but fails with resteasy-reactive with the following stacktrace:

`Caused by: java.lang.UnsupportedOperationException: Pending implementation
        at org.jboss.resteasy.reactive.common.jaxrs.RuntimeDelegateImpl.createConfigurationBuilder(RuntimeDelegateImpl.java:125)
        at jakarta.ws.rs.SeBootstrap$Configuration.builder(SeBootstrap.java:494)
        at org.jboss.resteasy.plugins.server.netty.NettyJaxrsServer.start(NettyJaxrsServer.java:85)
        at com.limetransit.fileencoder.worker.MockDispatcherResource.start(MockDispatcherResource.java:32)
        at io.quarkus.test.common.TestResourceManager$TestResourceEntryRunnable.run(TestResourceManager.java:492)
`

The reason can be found in [https://github.com/quarkusio/quarkus/blob/main/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/jaxrs/RuntimeDelegateImpl.java](url):
`    public SeBootstrap.Configuration.Builder createConfigurationBuilder() {
        // RR does not implement currently implement the bootstrapping API
        throw new UnsupportedOperationException("Pending implementation");
    }
`

Is there a roadmap for implementing this functionality?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
I very much doubt we will implement this

**Reply by FroMage** [reply]
Yeah, I'm not sure I see a use-case of running RR without Quarkus.

**Reply by sverkera** [reply]
My usecase is that I want to create a mock for REST endpoints that the service I'm testing is calling out to i.e. as a client.

For that I use a Resteasy Embedded container as described at https://docs.jboss.org/resteasy/docs/6.2.5.Final/userguide/html/ch41.html as a @QuarkusTestResource instanciated in a QuarkusTestResourceLifecycleManager. It works like a charm with Resteasy classic but if I use UndertowJaxrsServer that causes a classloader issue so that Resteasy reactive can't find the endpoints and NettyJaxrsServer throws exception as described above.

I.e. the scenario described here: https://quarkus.io/guides/rest-client#using-a-mock-http-server-for-tests
There it is suggested to use either Wiremock or Hoverfly but both are a royal pain in the behind to create a mock of a stateful service.

**Reply by geoand** [reply]
That is a use case we never really have come up with a satisfactory answer for unfortunately.

The bootstrap API is probably not the way we want to go with this, but likely we want something much simpler

**Reply by sverkera** [reply]
I asked about the bootstrap API since that is where NettyJaxrsServer threw exception. Possibly com.sun.net.httpserver.HttpServer would be easier to support, I didn't test that since the Resteasy documentation states "NOTE: This module is deprecated and should not be used."

**Reply by FroMage** [reply]
It would be nice to be able to declare endpoints from tests. ATM, we only can for Quarkus extension tests, I think. That would not be trivial to add, though.
